test/StpTests/PlayCheckerTests.cpp

```diff
@@ -20,13 +16,14 @@ class AlwaysValid : public rtt::ai::stp::Play {
     }
     virtual void calculateInfoForRoles() noexcept {
     }
+
+    bool isValidPlayToStart(rtt::world_new::World *world) noexcept override{return true;}
+    bool isValidPlayToKeep(rtt::world_new::World *world) noexcept override{return true;}
+    bool shouldRoleSkipEndTactic() override { return false;}
+
```

Comment:
This whole thing can be done cleaner by mocking these methods. Then you don't have to implement a new class for each situation

Reply:
Probably. I didn't write these tests, I'll make it an issue though